### DISCUSSION TITLE
Trying to install on a VM (x86_64)

### DISCUSSION THREAD

**Discussion by garbled1** [mcguirepr89/BirdNET-Pi#432] [General]
So I really would prefer to run this on a VM.  I only have a pi3b, and IMHO the hardware can barely keep up, but I have gobs and gobs of CPU at my disposal in x86 land.

So far, I've done the install, and gotten the basic stuff working, (on debian 11) but I'm stuck on 2 parts:

1) I can't get tflite working on the x86.  I downloaded one that was in the history of the repo, but it was linked against the wrong glibc.  I then tried to install the one off pip, but it's missing the flex stuff, and I can't figure out how to get that without building tflite-runtime from scratch.

2) gotty doesn't work.  It looks like you ship a modified pre-compiled version that has a path argument, that the base one does not?

Anyone else given this a shot?  Got anywhere?  Maybe have bins of those?

**Comment by ehpersonal38** [comment]
Did you check out the steps in the discussion here? 

https://github.com/mcguirepr89/BirdNET-Pi/discussions/418#discussioncomment-3072628

**Comment by Phaeton** [comment]
Also, take a look at #250. I was able to get gotty working, but ran into other issues. I was able to get my hands on a Pi4 so I gave up for the moment. However, would love to see this working on a VM or via Docker.

**Reply by garbled1** [reply]
Ok, yeah, that helped for gotty, Was able to get it fired up with that one.  Now I just need to figure out tensorflow.  I guess I'llhave to do the compile steps listed https://github.com/PINTO0309/TensorflowLite-flexdelegate there.

**Comment by mpfl** [comment]
I'm also interested in running this in an x86 Docker container. I have a Xeon-D server with a [Google Coral TPU](https://coral.ai/products/m2-accelerator-dual-edgetpu/), which can [accelerate Tensorflow Lite workloads](https://blog.tensorflow.org/2019/03/build-ai-that-works-offline-with-coral.html). It's working splendidly for object detection in video (I have a DIY smart doorbell), so would be interested to see it working for recognising birdcalls.

Of course, I only discovered Birdnet-Pi about an hour ago so I need to sort out a microphone first...

**Reply by epiech** [reply]
Any luck with that Coral TPU?

**Reply by mpfl** [reply]
Not at all. The BirdNET model uses AI functions that the Coral TPU can't execute. Gotta stick to CPU.

**Comment by chatainsim** [comment]
I'm also really interested in this.
I've only a pi 3B+ but a Proxmox server waiting for a new VM to be setuped.
Any progress ? I will try on my side. 
Thanks

**Comment by lloydbayley** [comment]


**Reply by chatainsim** [reply]
I've already tried to create a aarch64 VM but got really poor perf.
CPU emulation is really bad I think.
